src/taskgraph/run-task/run-task

```diff
@@ -1078,6 +1084,51 @@ def _display_python_version():
         b"Python version: "+ bytes(platform.python_version(), 'utf-8')
     )
 
+def _display_python_version_in_sub_process(subprocess: subprocess):
+    print_line(
+        b"setup",
+        b"pyhton --version: ",
+    )
+    try:
+        print_line(
+            b"setup",
+            subprocess.check_output(["python", "--version"])
+        )
+    except:
```

Comment:
Bug: Let's not catch all possible exceptions here. This will likely bite us in the future (e.g.: the machine is in a bad state, the script doesn't manage to shell out for whatever reason but it considers it's just Python missing).

Instead, I'd recommend catching `FileNotFoundError` which is the one raised: 

```
>>> subprocess.check_output(["python2", "--version"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 1024, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 1917, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'python2'
```